crates/agent-sql/src/publications.rs

```diff
@@ -208,6 +208,123 @@ pub async fn resolve_spec_rows(
     .await
 }
 
+#[derive(Debug)]
+pub struct Tenant {
+    pub name: String,
+    pub captures_quota: i32,
+    pub derivations_quota: i32,
+    pub materializations_quota: i32,
+    pub collections_quota: i32,
+    pub captures_used: i32,
+    pub derivations_used: i32,
+    pub materializations_used: i32,
+    pub collections_used: i32,
+}
+
+#[derive(Debug)]
+struct TenantFromDB {
+    pub name: String,
+    pub captures_quota: i32,
+    pub derivations_quota: i32,
+    pub materializations_quota: i32,
+    pub collections_quota: i32,
+    // These are Option<i64> because even if we add `coalesce(.., 0) to the query
+    // sqlx still infers them as nullable. So for our purposes assume None to be 0
+    pub captures_used: Option<i64>,
+    pub derivations_used: Option<i64>,
+    pub materializations_used: Option<i64>,
+    pub collections_used: Option<i64>,
+}
+
+impl From<TenantFromDB> for Tenant {
+    fn from(db: TenantFromDB) -> Self {
+        Tenant {
+            name: db.name,
+            captures_quota: db.captures_quota,
+            derivations_quota: db.derivations_quota,
+            materializations_quota: db.materializations_quota,
+            collections_quota: db.collections_quota,
+            captures_used: db.captures_used.unwrap_or(0) as i32,
+            derivations_used: db.derivations_used.unwrap_or(0) as i32,
+            materializations_used: db.materializations_used.unwrap_or(0) as i32,
+            collections_used: db.collections_used.unwrap_or(0).try_into().unwrap(),
+        }
+    }
+}
+
+pub async fn find_tenant_quotas(
```

Comment:
please add test coverage in `agent-sql/tests/publications.rs`

Reply:
done!